src/overlays/actors/ovl_En_Part/z_en_part.c

```diff
@@ -28,16 +30,123 @@ const ActorInit En_Part_InitVars = {
     (ActorFunc)EnPart_Draw,
 };
 
-#endif
+Vec3f D_80865900 = { 0.0f, 0.0f, 0.0f };
+
+static void (*D_8086590C[5])(EnPart*, PlayState*) = {
+    func_80865390, func_808654C4, 0x00000000, 0x00000000, 0x00000000,
+};
+
+void EnPart_Init(Actor* thisx, PlayState* play) {
+    EnPart* this = THIS;
+}
+
+void EnPart_Destroy(Actor* thisx, PlayState* play) {
+    EnPart* this = THIS;
+}
+
+void func_80865390(EnPart* this, PlayState* play) {
+    f32 randVar;
+
+    this->unk144 = 1;
+    this->actor.world.rot.y = (s16)(Rand_ZeroOne() * 20000.0f);
+    switch (this->actor.params) {
+        case 1:
+
+        case 4:
+            randVar = Rand_ZeroOne() * 17.0f;
+            this->unk146 += (s16)randVar + 5;
+            this->actor.velocity.y = (Rand_ZeroOne() * 5.0f) + 4.0f;
+            this->actor.gravity = (-0.6f) - (Rand_ZeroOne() * 0.5f);
+            this->unk14C = 0.15f;
+            break;
+
+        case 15:
+            this->actor.world.rot.y = this->actor.parent->shape.rot.y + 32768;
+            this->unk146 = 100;
+            this->actor.velocity.y = 7.0f;
+            this->actor.speedXZ = 2.0f;
+            this->actor.gravity = -1.0f;
+            break;
+    }
+}
+
+void func_808654C4(EnPart* this, PlayState* play) {
+    s16 tempVar;
+    Vec3f sp88;
+    Vec3f sp74;
+    s32 counter;
+    Vec3f sp64;
+
+    sp74 = D_80865900;
```

Comment:
This pattern usually is an inlined initializer for the variable, like this:
```suggestion
    Vec3f sp74 = { 0.0f, 0.0f, 0.0f };
    s32 counter;
    Vec3f sp64;
```

Doing this you'll need to remove the `D_80865900` you have at the start of the file, and probably move `D_8086590C` under this function.